core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -141,6 +141,22 @@ public T acquire(long time, TimeUnit unit) {
     }
   }
 
+  /**
+   * Acquires an object of type {@code T} from the pool. This operation is without blocking,
+   * if no resource is available. Each call of {@link #acquire()} should trigger to create
+   * a new object.
+   *
+   * @return a resource taken from the pool
+   */
+  public T acquireWithoutBlocking() {
+    // Try to take a resource without blocking
+    T resource = mResources.poll();
+    if (resource != null) {
+      return resource;
+    }
+    return createNewResource();
+  }
```

Comment:
Move the creation logic to caller side?
In this way, this `ResourcePool` provides a more isolated and independent function, just dealing with a pool of resources with different ways to acquire these resource (blocking or non-blocking)

```suggestion
  @Null
  public T acquireWithoutBlocking() {
    // Try to take a resource without blocking
    T resource = mResources.poll();
    if (resource != null) {
      return resource;
    }
    return null;
  }
```